### PR TITLE
[DRFT-253] Add notification service interface

### PR DESCRIPTION
**Pull request by eherget**
This is the start of KafkaProducer for sending baseline drift notifications to notifications service.

Optional to define the following env vars (default values are the current, correct values):
NOTIFICATION_SERVICE_TOPIC
NOTIFICATION_BUNDLE
NOTIFICATION_APP

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [x] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by eherget** [comment]
This is untested.
To generated a notification event, create a new EventDriftBaselineDetected instance with account_id, insights_id (system_id), system_check_in (in ISO-8601 format - it should already be in this format), display_name (of the system), and tags (if the exist).  Then, for each system-baseline comparison with a difference, call the add_drifted_baseline() method on the instance with baseline_id, baseline_name and the list of facts from the comparison.  The facts are not currently used as I didn't want to define it in the event payload only to have to change it later.  Once all the comparison reports are done, if any showed drift, the send_notification() method on the notification_service that was passed into _archive_profile() in archiver.py should be called with the instance of EventDriftBaselineDetected you created.
You'll notice the two methods defined in class _NotificationEventBase both do a json.dumps(xx).encode("utf-8") on the dict that gets passed in.  This is because the [send-notification spec](https://core-platform-apps.pages.redhat.com/notifications-docs/dev/apps-guide/send-notification.html) shows payload (7) and context (8) as being sent as json strings. Kinda odd, but that's what it says.

**Review by CodeHeeler — COMMENTED** [review]
Thanks for this work!  I'm still learning about kafka producers, but this looks good to me.

Question: is it sufficient to just have the bundle be insights and not have bundle_id like in the jira comment examples?

**Comment by eherget** [comment]
> Question: is it sufficient to just have the bundle be insights and not have bundle_id like in the jira comment examples?

Spec only shows bundle, not bundle_id. https://core-platform-apps.pages.redhat.com/notifications-docs/dev/apps-guide/send-notification.html

File: historical_system_profiles/notification_service_interface.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import json
+
+from datetime import datetime, timezone
+from kafka import KafkaProducer
+
+from historical_system_profiles import config
+
+NOTIFICATION_BUNDLE = "insights"
```
Comment (josejulio) [inline historical_system_profiles/notification_service_interface.py:8]: Current bundle in ci is `rhel` (previously it was `insights`)
https://ci.cloud.redhat.com/settings/notifications/rhel

Comment (petrblaho) [reply historical_system_profiles/notification_service_interface.py:8]: Is that value consistent across different envs (ci, qa, stage, prod)?

If not, we should make it to read from ENV.

Comment (eherget) [reply historical_system_profiles/notification_service_interface.py]: I have changed it to 'rhel' as per conversation with Thomas Heute

Comment (josejulio) [reply historical_system_profiles/notification_service_interface.py]: > Is that value consistent across different envs (ci, qa, stage, prod)?

Yes, it's consistent across envs.

Comment (eherget) [reply historical_system_profiles/notification_service_interface.py]: ...and it now also is an env variable

**Review by petrblaho — APPROVED** [review]
LGTM!

**Comment by eherget** [comment]
one last update to remove unused variables now done with env vars

**Comment by eherget** [comment]
sample event message should look like this:
{
  "bundle": "rhel",
  "application": "drift",
  "event_type": "drift-baseline-detected",
  "timestamp": "2021-04-20T16:23:17.386227+00:00",
  "account_id": "1212729",
  "events": [
    {
      "metadata": {},
      "payload": "{\"baseline_id\": \"35697c8b-29a8-41cc-b444-317f1921c4ae\", \"baseline_name\": \"rhel8-clone\"}"
    },
    {
      "metadata": {},
      "payload": "{\"baseline_id\": \"dc43ecfd-89ef-4aa6-8a6b-53f9c056fec5\", \"baseline_name\": \"new_baseline\"}"
    }
  ],
  "context": "{\"insights_id\": \"e3835949-5eb7-49cb-9b21-4d8433f43739\", \"system_check_in\": \"2021-04-19T20:40:40.087543\", \"display_name\": \"rhel8desktop\", \"tags\": [{\"value\": \"Jerome Marc\", \"key\": \"owner\"}, {\"value\": \"MBU\", \"key\": \"cost center\"}, {\"value\": \"\", \"key\": \"expiration date\"}, {\"value\": \"c2282417-a440-44f8-9d98-2a60cf8a52a7\", \"key\": \"inventory_id\"}, {\"value\": \"PMs\", \"key\": \"team\"}, {\"value\": \"rhel8desktop\", \"key\": \"display_name\"}, {\"value\": \"Management\", \"key\": \"business unit\"}, {\"value\": \"Production\", \"key\": \"environment\"}, {\"value\": \"RHEL Desktop\", \"key\": \"application\"}, {\"value\": \"empty\", \"key\": \"distribution list\"}, {\"value\": \"jmarc@redhat.com\", \"key\": \"contact\"}, {\"value\": \"jmarc\", \"key\": \"slack\"}, {\"value\": \"Sacramento\", \"key\": \"location\"}, {\"value\": \"United States\", \"key\": \"location\"}, {\"value\": \"California\", \"key\": \"location\"}]}"
}

**Pull request merged**